MathLib/LinAlg/GaussAlgorithm.h

```diff
@@ -33,8 +31,8 @@ namespace MathLib
  * the entries of A change! The solution for a specific
  * right hand side is computed by the method execute().
  */
-template <typename FLOAT_TYPE>
-class GaussAlgorithm : public MathLib::DenseDirectLinearSolver
+template <typename Matrix>
```

Comment:
@wenqing What is the idea using `Matrix` here? The existing GaussAlgorithm class internally uses Matrix.

Reply:
I also suggest keep `FLOAT_TYPE` in template parameters.

Reply:
@norihiro-w  
The reasons that I changed the template argument are
* There are two matrix classes in the source code:
   1.  https://github.com/ufz/ogs5/blob/master/MathLib/Matrix.h
   2.  https://github.com/ufz/ogs5/blob/master/FEM/matrix_class.h#L167
   The first one is used in some geometry calculation without unit test,
   and it was used as the internal  matrix in the direct linear solver.
   The second one is used in the local assembly. To enable the linear solver use the two type classes, the template argument is changed in this PR.   If the two matrix were merged into one in future, the template type would be removed.
* With FLOAT_TYPE, one can choose double, float or even complex type. Since we only need high  precision real numbers in the current FEM computation, I dropped this template argument.

Reply:
i see. thanks for the reply.